src/geohash_helper.c

```diff
@@ -280,3 +365,21 @@ int geohashGetDistanceIfInRectangle(double width_m,
     *distance = geohashGetDistance(x1, y1, x2, y2);
     return 1;
 }
+
+/* Check if a point is in a polygon using ray casting. */
+int geohashGetDistanceIfInPolygon(double x1, double y1, double *xy, double (*vertices)[2], int num_vertices, double *distance) {
```

Comment:
Do we need to distinguish whether the point is on the edge?

Reply:
Thank you , this is a good question. I did think about this when the implementation.

It is unlikely that a point will be on the an edge of a polygon to exact precision (except for the very clear cases) - so the BYPOLYGON might not return results when if we handle the "is a point on the edge" check.  


You can let me know what you think based on your experience with GIS applications

Reply:
Refer ray cast algorithm by W. Randolph Franklin https://github.com/tair-opensource/TairGis/blob/main/src/spatial/polyraycast.c#L157

Reply:
I will look into the linked algorithm and will get back to you here in the next day or two

Reply:
The current code in this function is based on PNPOLY - Point Inclusion in Polygon Test by W. Randolph Franklin (WRF)

https://wrfranklin.org/Research/Short_Notes/pnpoly.html

Reply:
Regarding points on the edge, doubles use different precision on ARM vs x86 so we get different rounding errors on different platforms (normal for floating point). We recently introduced an epsilon in one case here:

* #2063

Reply:
Thanks for sharing, @zuiderkwast .

If we really wanted to support including "points on the edge", we would have to use fabs / epsilon here as well.

I had posted [here](https://github.com/valkey-io/valkey/issues/1755#issuecomment-2819852982) that the BYRADIUS search only does not do this today when it checks if `distance > radius`:
```
int geohashGetDistanceIfInRadius(double x1, double y1, double x2, double y2, double radius, double *distance) {
    *distance = geohashGetDistance(x1, y1, x2, y2);
    if (*distance > radius) return 0;
    return 1;
}
```

So, I don't think we need to handle this for the BYPOLYGON search

Reply:
Agree, let's not do it now.

In #2063 we did it only for the special case of finding a point with zero radius from itself.